pkg/types/types.go

```diff
@@ -69,34 +69,36 @@ type NetConf struct {
 
 // Result is what gets returned from the plugin (via stdout) to the caller
 type Result struct {
-	IP4 *IPConfig `json:"ip4,omitempty"`
-	IP6 *IPConfig `json:"ip6,omitempty"`
-	DNS DNS       `json:"dns,omitempty"`
+	Interfaces []*Interface `json:"interfaces,omitempty"`
+	IP         []*IPConfig  `json:"ip,omitempty"`
```

Comment:
@steveeJ requested "IP" in https://github.com/containernetworking/cni/pull/145#discussion-diff-71392017